mlflow/pipelines/steps/split.py

```diff
@@ -63,8 +63,11 @@ def _get_split_df(input_df, hash_buckets, split_ratios):
 
 def _hash_pandas_dataframe(input_df):
     from pandas.util import hash_pandas_object
+    from pandarallel import pandarallel
 
-    return hash_pandas_object(input_df.applymap(_make_elem_hashable))
+    pandarallel.initialize(progress_bar=False)
```

Comment:
I'm not sure we want to introduce this package but don't have a strong feeling about it. We might move to Spark Dataframe at some point but it could take quite a while.

Do you mind do more thorough measurements on the performance gains by varying the data along these 2 dimensions?
- data size: 10 MB / 1GB
- num of columns: 10 / 1000
- machine type: local laptop / Databricks
- number of cores (can you control this?): 2 / 4 / 8 / 16

If we get good speedup (2x or more) on most testing combinations, it makes sense to me to use the package.
Thanks a lot, @prithvikannan !

Reply:
I don't think we should introduce new dependencies unless it's necessary / really hard to do without the new dependency. Are there any alternatives? What about a multithreaded approach that iterates over different portions of the dataset?

Reply:
In any case, we want MLP to run on Databricks by default in MLflow 2.0. To do that, we would need this dependency to be present in MLR. Is it present in MLR?

Reply:
> What about a multithreaded approach that iterates over different portions of the dataset?

I believe that is what pandarallel does under the hood. [This stackoverflow post](https://stackoverflow.com/questions/26784164/pandas-multiprocessing-apply#:~:text=A%20more%20generic%20version%20based%20on%20the%20author%20solution%2C%20that%20allows%20to%20run%20it%20on%20every%20function%20and%20dataframe%3A) outlines a way to do this with just python native multiprocessing.